### PR TITLE
Clean AAG/Dashboard cards.  

### PR DESCRIPTION
**Pull request by dereksmart**
**This is not a functional change.** It is a refactor of all of the Dashboard cards to:

- Use the actions from `state/settings/`, rather than the old (deprecated) `state/modules/` actions.  
- Clean up all eslint warnings, including registering es6 class components. (Cleaned up around 60 warnings)
- Make this more dry. We're now passing down the settings actions through so that each dash card does not need to register it's own.  
- Remove unused code.  For example, we were no longer using the `DashSiteVerify` component, so I removed it completely.  

We should be moving to completely deprecate the use of `state/modules/` actions, like `activateModule`, `deactivateModule`, etc... and instead module activation should be done through `state/settings/`.  This is one step in that direction, as there are still other places that use the old actions.  Submitting this now so that it doesn't get too crazy.  

**TO TEST:** 
- Do everything imaginable in the Dashboard area.  
- Do it again, and in a different state (different plan, etc...)
- Activate/deactivate ALL of the modules listed there
- Test the connection stuff
- Test it in non-admin mode

### CODE REVIEW

**Comment by eliorivero** [comment]
- GUI tests are failing https://travis-ci.org/Automattic/jetpack/jobs/232880680
- there are some eslint errors at
    ```
    client/state/at-a-glance/index.js
    client/at-a-glance/backups.jsx
    client/at-a-glance/connections.jsx
    ```

**Comment by oskosk** [comment]
Such a diff!

File: _inc/client/at-a-glance/index.jsx [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
 import { translate as __ } from 'i18n-calypso';
 import analytics from 'lib/analytics';
 
 /**
  * Internal dependencies
  */
+import { ModuleSettingsForm as moduleSettingsForm } from 'components/module-settings/module-settings-form';
 import DashSectionHeader from 'components/dash-section-header';
-import DashStats from './stats';
+import DashStats from './stats/index.jsx';
```
Comment (oskosk) [inline _inc/client/at-a-glance/index.jsx]: Not quite clear to me why we need the extra `/index.jsx` here

File: _inc/client/at-a-glance/akismet.jsx [hunk 2]
```diff
@@ -158,16 +138,8 @@ DashAkismet.propTypes = {
 export default connect(
 	( state ) => {
 		return {
-			getAkismetData: () => _getAkismetData( state ),
-			isModuleActivated: ( module_name ) => _isModuleActivated( state, module_name ),
-			getSitePlan: () => getSitePlan( state )
+			getAkismetData: getAkismetData( state ),
+			getSitePlan: getSitePlan( state )
```
Comment (oskosk) [inline _inc/client/at-a-glance/akismet.jsx]: These two prop names are confusing now that they're no longer functions. `akimestData` and `getSitePlan` would be clearer

File: _inc/client/at-a-glance/index.jsx (hunk 1 above)
Comment (dereksmart) [inline _inc/client/at-a-glance/index.jsx]: We technically don't but my text editor was complaining about it not being there /shrug

**Comment by dereksmart** [comment]
@eliorivero 

> there are some eslint ~errors~ warnings at
client/state/at-a-glance/index.js
client/at-a-glance/backups.jsx
client/at-a-glance/connections.jsx

Fixed these.  The warnings coming from `at-a-glance/connections.jsx` was about us using the wrong gridicon size, `jsx-gridicon-size`, so I just disabled the rule rather than changing the design we already had.

**Comment by eliorivero** [comment]
captura de pantalla 2017-05-16 a las 17 58 35

User connection is broken for me.

**Comment by dereksmart** [comment]
> However, user connection is broken for me.

@eliorivero I can't reproduce.  Can you see if this happens for you in other branches as well? Or after you cycle the connection?  Or can you look at what props are being passed to that component?  Thanks!

**Comment by oskosk** [comment]
LGTM, couldn't reproduce the error with the connection dashboard item

**Comment by eliorivero** [comment]
Connection error is gone now. Not sure why I was experiencing this before. It's ok so will merge.

**Pull request merged**